src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,190 @@
+        if (str.split(" ").length == 1) {
+            //check if task if valid
+            if (str.startsWith("todo") || str.startsWith("deadline") || str.startsWith("event")) {
+                throw new EmptyDescriptionException(str);
+            } else { //check for invalid task
+                throw new InvalidTaskException();
+            }
+        }
+
+        //task that contain more than 1 word.
+        //check for invalid task. otherwise, add the correct task to the list.
+        else {
+            if (!str.startsWith("todo") && !str.startsWith("deadline") && !str.startsWith("event")) {
+                throw new InvalidTaskException();
+            }
+            else {
+                Task task;
+                if (str.startsWith("todo")) {
+                    task = new ToDos(str.substring(5));
+                    list.add(task);
+                } else if (str.startsWith("deadline")) {
+                    String[] message = str.split("/by ");
+                    if (message.length == 1) {
+                        throw new InvalidTaskException();
+                    } else {
+                        task = new Deadline(message[0].substring(9), message[1]);
+                        list.add(task);
+                    }
+                } else {
+                    String[] message = str.split("/at ");
+                    if (message.length == 1) {
+                        throw new InvalidTaskException();
+                    } else {
+                        task = new Events(message[0].substring(6), message[1]);
+                        list.add(task);
+                    }
+                }
+
+                System.out.println("Got it. I've added this task.");
+                System.out.println(list.get(list.size() - 1));
+                System.out.println("Now you have " + list.size()
+                        + (list.size() == 1 ? " task in the list" : " tasks in the list."));
+                return task;
+            }
+        }
+    }
+
+    /**
+     * A method to delete a certain in a TaskList.
+     * @param str User inputs which indicates detail of task to be deleted.
+     * @return Task after it is deleted.
+     * @throws OutOfBoundException Thrown when user gives a task number that is outside of the range of TaskList.
+     */
+    public Task deleteTask(String str) throws OutOfBoundException {
+        String i = str.substring(str.length()-1);
+        int index = Integer.parseInt(i);
+
+        if (index < 0 || index > list.size()) {
+            throw new OutOfBoundException();
+        } else {
+            Task task = list.get(index - 1);
+            list.remove(index - 1);
+            System.out.println("Noted. I've removed this task: ");
+            System.out.println(task);
+            System.out.println("Now you have " + list.size()
+                    + (list.size() == 1 ? " task in the list" : " tasks in the list."));
+            return task;
+        }
```

Comment:
since the if statement to throw the exception is a guard clause already, could the else statement be removed?